### PR TITLE
Reload/detect native decoding after shutdown

### LINKED ISSUE #259: Native decoding - sometimes supported, sometimes not
Sometimes the VTR log tells me that native decoding is supported, and sometimes it tells me it is not:

```
2019-10-03T09:29:49     WARNING    Native decoding supported!!! (Windows, 64bit)
```
```
2019-10-03T09:51:34     WARNING    Native decoding not supported. (Windows, 64bit)
```

Could the test mechanism be unreliable in any way?

### PR DESCRIPTION
**Pull request by tomchadwin**
Fixes: #259

Please see whether you think this possible fix is correct.

### CODE REVIEW

**Comment by tomchadwin** [comment]
Seems to behave correct;y for me, **but** one time reloading VTR after this change, using the plugin reloader plugin, I got a full QGIS crash:

```
Stack Trace


QBitArray::QBitArray :
QgsVirtualLayerDefinition::query qgsvirtuallayerdefinition.h:120
QgsExpression::functionIndex qgsexpression.cpp:179
QgsExpression::isFunctionName qgsexpression.cpp:171
meth_QgsExpression_isFunctionName sip_corepart3.cpp:216815
PyMethodDef_RawFastCallKeywords :
PyMethodDef_RawFastCallKeywords :
PyEval_EvalFrameDefault :
PyEval_EvalCodeWithName :
PyFunction_FastCallDict :
PyException_SetTraceback :
PyEval_EvalFrameDefault :
PyEval_EvalCodeWithName :
PyMethodDef_RawFastCallKeywords :
PyEval_EvalFrameDefault :
PyEval_EvalCodeWithName :
PyImport_Import :
PyImport_Import :
PyMethodDef_RawFastCallDict :
PyException_SetTraceback :
PyEval_EvalFrameDefault :
PyEval_EvalCodeWithName :
PyMethodDef_RawFastCallKeywords :
PyEval_EvalFrameDefault :
PyMethodDef_RawFastCallKeywords :
PyEval_EvalFrameDefault :
PyMethodDef_RawFastCallKeywords :
PyEval_EvalFrameDefault :
PyMethodDef_RawFastCallKeywords :
PyEval_EvalFrameDefault :
PyFunction_FastCallDict :
PyObject_CallMethodIdObjArgs :
PyObject_CallMethodIdObjArgs :
PyUnicode_FindChar :
PyImport_ImportModuleLevelObject :
PyDict_GetItemString :
PyException_SetTraceback :
PyException_SetTraceback :
PyEval_EvalFrameDefault :
PyEval_EvalCodeWithName :
PyMethodDef_RawFastCallKeywords :
PyEval_EvalFrameDefault :
PyEval_EvalCodeWithName :
PyFunction_FastCallDict :
PyObject_CallMethodIdObjArgs :
PyObject_CallMethodIdObjArgs :
PyImport_ImportModuleLevelObject :
PyDict_GetItemString :
PyMethodDef_RawFastCallKeywords :
PyMethodDef_RawFastCallKeywords :
PyEval_EvalFrameDefault :
PyEval_EvalCodeWithName :
PyFunction_FastCallDict :
PyErr_NoMemory :
PyEval_EvalFrameDefault :
PyEval_EvalCodeWithName :
PyImport_Import :
PyImport_Import :
PyMethodDef_RawFastCallDict :
PyException_SetTraceback :
PyEval_EvalFrameDefault :
PyEval_EvalCodeWithName :
PyMethodDef_RawFastCallKeywords :
PyEval_EvalFrameDefault :
PyMethodDef_RawFastCallKeywords :
PyEval_EvalFrameDefault :
PyMethodDef_RawFastCallKeywords :
PyEval_EvalFrameDefault :
PyMethodDef_RawFastCallKeywords :
PyEval_EvalFrameDefault :
PyFunction_FastCallDict :
PyObject_CallMethodIdObjArgs :
PyObject_CallMethodIdObjArgs :
PyUnicode_FindChar :
PyImport_ImportModuleLevelObject :
PyDict_GetItemString :
PyException_SetTraceback :
PyException_SetTraceback :
PyEval_EvalFrameDefault :
PyEval_EvalCodeWithName :
PyMethodDef_RawFastCallKeywords :
PyEval_EvalFrameDefault :
PyEval_EvalCodeWithName :
PyFunction_FastCallDict :
PyObject_CallMethodIdObjArgs :
PyObject_CallMethodIdObjArgs :
PyImport_ImportModuleLevelObject :
PyDict_GetItemString :
PyMethodDef_RawFastCallKeywords :
PyMethodDef_RawFastCallKeywords :
PyEval_EvalFrameDefault :
PyEval_EvalCodeWithName :
PyFunction_FastCallDict :
PyErr_NoMemory :
PyEval_EvalFrameDefault :
PyEval_EvalCodeWithName :
PyImport_Import :
PyImport_Import :
PyMethodDef_RawFastCallDict :
PyException_SetTraceback :
PyEval_EvalFrameDefault :
PyEval_EvalCodeWithName :
PyMethodDef_RawFastCallKeywords :
PyEval_EvalFrameDefault :
PyMethodDef_RawFastCallKeywords :
PyEval_EvalFrameDefault :
PyMethodDef_RawFastCallKeywords :
PyEval_EvalFrameDefault :
PyMethodDef_RawFastCallKeywords :
PyEval_EvalFrameDefault :
PyFunction_FastCallDict :
PyObject_CallMethodIdObjArgs :
PyObject_CallMethodIdObjArgs :
PyUnicode_FindChar :
PyImport_ImportModuleLevelObject :
PyDict_GetItemString :
PyMethodDef_RawFastCallKeywords :
PyMethodDef_RawFastCallKeywords :
PyEval_EvalFrameDefault :
PyEval_EvalCodeWithName :
PyFunction_FastCallDict :
PyErr_NoMemory :
PyEval_EvalFrameDefault :
PyMethodDef_RawFastCallKeywords :
PyEval_EvalFrameDefault :
PyMethodDef_RawFastCallKeywords :
PyEval_EvalFrameDefault :
PyMethodDef_RawFastCallKeywords :
PyEval_EvalFrameDefault :
PyMethodDef_RawFastCallKeywords :
PyEval_EvalFrameDefault :
PyFunction_FastCallDict :
PyMethodDef_RawFastCallDict :
PyObject_Call :
PyInit_QtCore :
PyInit_QtCore :
PyInit_QtCore :
PyInit_QtCore :
PyInit_QtCore :
QMetaObject::activate :
QAction::activate :
QAbstractButton::click :
QAbstractButton::mouseReleaseEvent :
QToolButton::mouseReleaseEvent :
QWidget::event :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify qgsapplication.cpp:403
QCoreApplication::notifyInternal2 :
QApplicationPrivate::sendMouseEvent :
QSizePolicy::QSizePolicy :
QSizePolicy::QSizePolicy :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify qgsapplication.cpp:403
QCoreApplication::notifyInternal2 :
QGuiApplicationPrivate::processMouseEvent :
QWindowSystemInterface::sendWindowSystemEvents :
QEventDispatcherWin32::processEvents :
UserCallWinProcCheckWow :
DispatchMessageWorker :
QEventDispatcherWin32::processEvents :
qt_plugin_query_metadata :
QEventLoop::exec :
QCoreApplication::exec :
main main.cpp:1567
BaseThreadInitThunk :
RtlUserThreadStart :




QGIS Info
QGIS Version: 3.8.3-Zanzibar
QGIS code revision: 685d8b15d2
Compiled against Qt: 5.11.2
Running against Qt: 5.11.2
Compiled against GDAL: 2.4.1
Running against GDAL: 2.4.1



System Info
CPU Type: x86_64
Kernel Type: winnt
Kernel Version: 6.1.7601
```

Any idea what this dump tells us?

File: plugin/util/mp_helper.py [hunk 1]
```diff
@@ -106,6 +105,8 @@ def unload_lib():
 
 
 def native_decoding_supported() -> bool:
+    global _native_lib_handle
+    _native_lib_handle = load_lib()
```
Comment (mnboos) [inline plugin/util/mp_helper.py:109]: I don't want to load the lib here. This function should only report whether the lib was loaded or not.

Comment (tomchadwin) [reply plugin/util/mp_helper.py:109]: OK. Hopefully this gives you enough info on the underlying cause.

**Comment by mnboos** [comment]
Fixed

**Pull request closed without merging**